src/IssueViz.Security/AnalysisSeverityToPriorityConverter.cs

```diff
@@ -0,0 +1,82 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security
+{
+    internal interface IAnalysisSeverityToPriorityConverter
+    {
+        /// <summary>
+        /// Converts <see cref="AnalysisIssueSeverity"/> into high/medium/low priority
+        /// </summary>
+        string Convert(AnalysisIssueSeverity severity);
+
+        /// <summary>
+        /// Converts high/medium/low priority into <see cref="AnalysisIssueSeverity"/>
+        /// </summary>
+        AnalysisIssueSeverity Convert(string priority);
+    }
+
+    internal class AnalysisSeverityToPriorityConverter : IAnalysisSeverityToPriorityConverter
+    {
+        private readonly IDictionary<string, AnalysisIssueSeverity> map = new Dictionary<string, AnalysisIssueSeverity>
+        {
+            {"high", AnalysisIssueSeverity.Blocker},
+            {"medium", AnalysisIssueSeverity.Major},
+            {"low", AnalysisIssueSeverity.Minor}
```

Comment:
Hotspots and Issue have different Severities and Types so we can't conveniently represent a hotspot as an 'IAnalysisIsue'.
We either need to have a re-think about how to use the visualization/issue interfaces so they work nicely with hotspots and issues, or we hack around with the hotspots representation to pretend they are issues too. This looks like the second approach. We should have a chat about the best way to address this problem.

Reply:
I think a clean solution will be reasonably straightforward as far as the taggers go. As far as I remember the taggers are only interested in location, and whether the location is primary or secondary.

The data sources should be ok too since the Error List data source only cares about issues, and the hotspot data source only cares about hotspots (at least at the moment).

That leaves the flow pane which shows an icon based on the severity. With a bit of luck we'll be able to get away with tweaking the visualization interfaces so they don't need to use the hotspot/issue severity directly.